Compile error - macOS 10.14 Mojave

Hi Micmac Team and congratulations for this tools !

I'm trying to compile MicMac for MacOS Mojave (10.14) and share binaries with community.
But, I have some errors...
No problem to install dependencies with brew.

Last commands :
$ cmake ../
$ make clean
$ make install -j6

First output error :
-- -----> OpenGL found
-- -----> using Qt 5.12.0
-- -----> Compile with c++11 support
-- -- FLAG:  -std=gnu++11
-- -- Git executable found
-- Derniere revision git de MicMac : v1.0.beta13-360-g3fe2f447e
-- Configuring done
-- Generating done
-- Build files have been written to: /usr/local/micmac/build
edo:build laurent$ make install -j6
[  0%] Built target ScriptCalib
[  0%] Built target EPExeRechCibleInit
[  1%] Built target EPExeCalibInit
[  1%] Built target poisson
[  1%] Generating ../../include/StdAfx.h.pch
[  2%] Built target ANN
In file included from /usr/local/micmac/include/StdAfx.h:28:
In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/memory:653:
In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/typeinfo:61:
In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/exception:82:
In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/cstdlib:86:
/Library/Developer/CommandLineTools/usr/include/c++/v1/stdlib.h:94:15: fatal error: 'stdlib.h' file not found
#include_next <stdlib.h>

Last output errors (and warnings) before die :
In file included from /usr/local/micmac/CodeExterne/ANN/src/kd_search.h:30:
/usr/local/micmac/CodeExterne/ANN/src/pr_queue_k.h:103:4: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
                        register int i;
                        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:174:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register ANNdist dist;                          // distance to data point
        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:175:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register ANNcoord* pp;                          // data coordinate pointer
        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:176:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register ANNcoord* qq;                          // query coordinate pointer
        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:177:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register ANNdist min_dist;                      // distance to k-th closest point
        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:178:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register ANNcoord t;
        ^~~~~~~~~
/usr/local/micmac/CodeExterne/ANN/src/kd_search.cpp:179:2: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
        register int d;
        ^~~~~~~~~
7 warnings generated.
[  2%] Linking CXX static library libANN.a
[  2%] Built target ANN
[  2%] Linking CXX static library libpoisson.a
[  2%] Built target poisson
make: *** [all] Error 2

...and die !

Thanks for help.

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,9 @@ matrix:
   - os: osx
     osx_image: xcode9.3
     compiler: clang
+  - os: osx
+    osx_image: xcode10.2
+    compiler: clang
 
 before_install:
 - if [ "${TRAVIS_OS_NAME}" = "linux" ]; then sudo apt-get update -qq && sudo apt-get

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -257,7 +257,7 @@ if(${WITH_CPP11})
 	if(NOT MSVC)
 		message(STATUS "-----> Compile with c++11 support")
 		# support c++11
-		if(NOT DARWIN) #TODO: if(NOT clang), the "precomopiled header bug" it tries to fix, is clang related, not osx related   
+		if(NOT DARWIN) #TODO: if(NOT clang) => the "precompiled header bug" it tries to fix is clang related, not osx related   
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 		ELSE()
 		# MODIF MPD, reverse Gregoire, car gnu++11 mulitplie par 6 le temps de compilation

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -175,7 +175,11 @@ endif()
 install(TARGETS ${libElise} ARCHIVE DESTINATION ${BUILD_PATH_LIB})
 
 if(${WITH_HEADER_PRECOMP} AND (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_IS_CLANG))
-	enable_precompiled_headers_GCC(StdAfx.h ${libElise} "-fPIC -std=gnu++11")
+	if(CMAKE_OSX_SYSROOT) # for macOS > 10.14, it contains the path to the SDK/headers. checking this fixes issue #113
+		enable_precompiled_headers_GCC(StdAfx.h ${libElise} "-fPIC -std=gnu++11 -isysroot ${CMAKE_OSX_SYSROOT}")
+	else()
+		enable_precompiled_headers_GCC(StdAfx.h ${libElise} "-fPIC -std=gnu++11")
+	endif()
 	set(precompiled_header ${precompiled_header} PARENT_SCOPE)
 endif()
 

